python/feathub/processors/flink/table_builder/flink_table_builder.py

```diff
@@ -549,6 +568,49 @@ def _get_table_from_sliding_feature_view(
             *[native_flink_expr.col(field) for field in output_fields]
         )
 
+    def _get_table_from_sql_feature_view(
+        self, feature_view: SqlFeatureView
+    ) -> NativeFlinkTable:
+        for table_name in self._get_all_registered_table_names():
+            if (
+                table_name in self.t_env.list_temporary_views()
+                or table_name in self._tables_being_built
+            ):
+                continue
+            self.t_env.create_temporary_view(
+                table_name, self._get_table(self.registry.get_features(table_name))
+            )
+
+        result_table = self.t_env.sql_query(feature_view.sql_statement)
+
+        if feature_view.timestamp_field is not None:
+            result_stream = self.t_env.to_data_stream(result_table)
+
+            # TODO: add support for max_out_of_orderness in SqlFeatureView.
+            result_table = self.t_env.from_data_stream(
+                result_stream,
+                define_watermark(
+                    t_env=self.t_env,
+                    flink_schema=to_flink_schema(feature_view.schema),
+                    max_out_of_orderness=timedelta(0),
+                    timestamp_field=feature_view.timestamp_field,
+                    timestamp_format=feature_view.timestamp_format,
+                    timestamp_field_dtype=feature_view.schema.get_field_type(
+                        feature_view.timestamp_field
+                    ),
+                ),
+            )
+
+        return result_table
+
+    # TODO: Add public API to get all registered features in a Registry instance
```

Comment:
Since the long term plan is to get the source tables by parsing the SQL statement instead of registering all tables found in the registry, it seems better to remove this TODO.